File: SkillSync.Client/src/app/shared/entities/responses/freelancer-reviews-dto.ts
import { GetFreelancerReviewDto } from "./get-freelancer-review-dto";
import { ProjectReviewDataDto } from "./project-review-data-dto";
import { ReviewStatisticsDto } from "./review-statistics-dto";

export interface FreelancerReviewsDto {
    usersProfilePicture: { [key: string]: string };
    projects: { [key: string]: ProjectReviewDataDto };
    reviews: GetFreelancerReviewDto[];
    reviewStatistics: ReviewStatisticsDto;
}

File: SkillSync.WebServer/SkillSync.Domain/DTOs/Review/FreelancerReviewsDto.cs
namespace SkillSync.Domain.DTOs.Review
{
    public class FreelancerReviewsDto
    {
        public Dictionary<string, string> UsersProfilePicture { get; set; }

        public Dictionary<Guid, ProjectReviewDataDto> Projects { get; set; }

        public List<GetFreelancerReviewDto> Reviews { get; set; }

        public ReviewStatisticsDto ReviewStatistics { get; set; }
    }

    public class ProjectReviewDataDto
    {
        public string Name { get; set; }

        public string Picture { get; set; }
    }
}